client/src/components/user/dashboard.js

```diff
@@ -72,10 +89,29 @@ class Dashboard extends React.Component {
     this.setState({ subscriptions });
   };
 
-  setEditingSubscription = async (editingSubscription) => {
-    await this.setState({ editingSubscription });
+  async handleDashboardChange(newView) {
+    const { mainComponentView } = this.state;
+
+    if (mainComponentView === 'dashboardCalendar' || mainComponentView === 'subscriptionDetail') {
+      return this.setState({ mainComponentView: newView });
+    }
+
+    const modal = new Modal(document.getElementById('myModal'));
+    modal.show();
+
+    const userInput = await waitUserClick();
+    if (userInput === 'next') {
+      await this.setState({ mainComponentView: newView });
+    } else if (userInput === 'close') {
+      await this.setState({ mainComponentView });
+    }
+    return modal.hide();
   }
 
+  setMainComponentView = async (newView) => {
+    await this.setState({ mainComponentView: newView });
```

Comment:
same as above here and everywhere else that you're doing it, i don't think you need the async await when using the `this.setState()` cal